report/string_set.go

```diff
@@ -54,33 +54,35 @@ func (s StringSet) Intersection(b StringSet) StringSet {
 // Add adds the strings to the StringSet. Add is the only valid way to grow a
 // StringSet. Add returns the StringSet to enable chaining.
 func (s StringSet) Add(strs ...string) StringSet {
+	r := s.Copy()
```

Comment:
The change to `StringSet` is really unrelated to this PR. I reckon we should ditch it.

AFAICT all the uses of `StringSet.Add/Remove` are in contexts where we've just created a fresh node and are populating it. We may want to avoid all this continued allocation/copying for that. But perhaps by having different methods, to avoid confusion with `Add/Remove` methods elsewhere that *do* copying.